### PR TITLE
Add alignment options to flow container

### LINKED ISSUE #4453: Please add Right-to-left alignment to HFlowContainer
### Describe the project you are working on

Any project that users HFlowContainer.

### Describe the problem or limitation you are having in your project

HFlowContainer appears to allow left-to-right alignment:

```
--------
a b c
d  e
--------
```

But does not appear to allow:

```
---------
    a b c
     d  e
---------
```

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Allow right justification like this:

```
---------
    a b c
     d  e
---------
```

NB: I initially used the term 'right-to-left' here, but that term doesn't appear to imply what I had intended.

### Is there a reason why this should be core and not an add-on in the asset library?

It doesnt make sense to support HFlowContainer left-to-right but not allow right-to-left.

### PR DESCRIPTION
**Pull request by 98teg**
This PR adds an option to modify the alignment of flow containers. I have used BoxContainer as a reference for this implementation. Here is a video showcasing the results:
https://www.youtube.com/watch?v=SGOTBxSrCOw
It would solve this proposal: https://github.com/godotengine/godot-proposals/issues/4453
If this gets approved I would be glad to backport it to Godot 3.x

*Bugsquad edit:*
- Closes https://github.com/godotengine/godot-proposals/issues/4453

### CODE REVIEW

File: scene/gui/flow_container.cpp [hunk 1]
```diff
@@ -152,6 +152,28 @@ void FlowContainer::_resort() {
 			line_data = lines_data[current_line_idx];
 		}
 
+		// The first child of each line adds the offset caused by the aligment,
+		// but only if the line doesn't contain a child that expands
```
Comment (kleonc) [inline scene/gui/flow_container.cpp:156]: [Comments are supposed to end with a period](https://docs.godotengine.org/en/latest/community/contributing/code_style_guidelines.html#comment-style-guide).
```suggestion
		// The first child of each line adds the offset caused by the alignment,
		// but only if the line doesn't contain a child that expands.
```

**Review by kleonc — APPROVED** [review]
LGTM.

BTW `AspectRatioContainer`, `BoxContainer`, and now `FlowContainer` all have the same `enum Alignment` duplicated. [There are also some alignment enums in the @GlobalScope](https://github.com/godotengine/godot/blob/0d04e7ec914ff609dfbdfff3dca04ed12a21e600/doc/classes/%40GlobalScope.xml#L1408-L1467). So I wonder if maybe these alignments should be moved to the global scope or if maybe the other alignments should be removed from the global scope (and moved to the appropriate classes).
Nonetheless that's for a potential separate PR of course. In here it's fine as is.

**Review by KoBeWi — APPROVED** [review]
CC @Geometror

**Review by Geometror — APPROVED** [review]
Looks like everything works as it should :)

Regarding the `Alignment` enum: That is something which struck me a few times in the past, I think the Horizontal-/VerticalAlignment enums could be consolidated into one `Alignment` enum since both have essentially the same values and it seems like you have always enough context to know whether it is horizontal or vertical. (Most method and property names include this direction prefix). 
But either way, that is material for another PR.

**Comment by akien-mga** [comment]
Thanks! And congrats for your first merged Godot contribution :tada:

**Pull request merged**